Draft-Accepted/RFCNNNN-PathUpdateForPackageInstall.md

```diff
@@ -0,0 +1,189 @@
+
+   - Would it be reasonable to disable the feature in PowerShell but not in CMD?
+   - [dongbow] IMHO, it should be OK for us to honor this key too.
+     It seems not reasonable to have a separate way to disable this feature in PowerShell while keeping the feature on in CMD.
+
+   ```c
+   if (DosErr == MSG_DIR_BAD_COMMAND_OR_FILE) {
+       if (Feature_WingetCmdNotFound::IsEnabled()) {
+           if (fIsInteractiveForExtensionPurposes && dwInteractiveExtensionLevel >= 1) {
+                // handle winget not found error specially
+           }
+       }
+   } else { ... }
+   ```
+
+   - **_[Followed up with WT]_** on `InteractiveExtensionLevel`
+     - WT response:
+       > - it was our "get out of jail" card in case we legitimately broke something someone cared about.
+           If you have a risk of breakage due to this new behavior, users may want a way to turn it off.
+       > - we haven't heard that we have broken anyone. But the new changes don't have so much market adoption yet.
+       > - feel free to follow it. However, other things in PowerShell DON'T follow it today
+     - To summarize:
+       - It's for turning off all interactive improvements they made to CMD, not exclusive to the PATH refreshing feature
+       - PS can choose to use it, but it's not used by PowerShell for any other things.
+
+   - **[PS REVIEW COMMENT]** We will use the same registry key for the list of package manager but **Will NOT** hornor this key: `InteractiveExtensionLevel`,
+     because that key is not exclusively for the package manager detection feature.
+
+   - **[PS REVIEW COMMENT]** _"Do we need to have a PowerShell way to disable the feautre?"_ -- maybe NOT, just depend on clearing the list.
+     - WT response:
+       > clearing the list will change it for cmd as well, of course. A windows upgrade might restore part of the list (the three entries we know about)
+       > - we migrate `InteractiveExtensionLevel` and the package manager info from build to build
+       > - however, I don't know if migration manifests account for deletions of keys. that's where it gets weird for me, and that's where I don't actually know
+     - [dongbow && jason]'s thoughts on this comment
+       - It's unlikely that a user wants to use different settings for that list between PowerShell and CMD.
+       - Windows upgrade shouldn't mess up with the changes a user made to those registry keys. It sounds like a bug if it does.
+       - Even if deleted key will be restored with the default value by windows upgrade, clearing the list but keeping the key around should be fine with windows upgrade.
+
+4. What about `Start-Process`, should we do the detection within that cmdlet too?
+
+   - **[PS REVIEW COMMENT]** No, the feature is only for the native command processor.
+
+   - **_[WT comments]_** CMD only covers EXE, but `scoop` is a `.ps1` or `.bat` file, so CMD doesn't see it. PowerShell may want to broaden your support to account for any command.
```

Comment:
Only EXE? There is above:
> Batch files **run from an interactive shell** will detect package managers as well,
> and this is going to help things like `install_all_my_packages.bat`.

So pwsh should follow this too. Although perhaps the only way to make it simple is to specify these bat files in the registry.